src/Adapter/MSTest.CoreAdapter/Execution/TestMethodInfo.cs

```diff
@@ -137,9 +151,13 @@ public virtual TestResult Invoke(object[] arguments)
 
             using (LogMessageListener listener = new LogMessageListener(this.TestMethodOptions.CaptureDebugTraces))
             {
+                var testElement = new UnitTestElement(new TestMethod(this)) { TestId = this.TestId };
+
                 watch.Start();
                 try
                 {
+                    this.TestExecutionRecorder.RecordStart(testElement);
```

Comment:
@jayaranigarg this is where I moved firing the start event to. This happens right before we invoke test setup/execute/cleanup. This ensures we fire an event for each iteration of a data driven tests. It also happens after the custom test execution extension is called which means if someone writes an extension to run the same test 10 times we will fire 10 start/end events (which IMO is expected).